crates/firewheel-nodes/src/freeverb/mod.rs

```diff
@@ -0,0 +1,302 @@
+                FreeverbNodePatch::SmoothSeconds(value) => {
+                    self.room_size
+                        .set_smooth_seconds(value, proc_info.sample_rate);
+                    self.width.set_smooth_seconds(value, proc_info.sample_rate);
+                    self.damping
+                        .set_smooth_seconds(value, proc_info.sample_rate);
+                }
+            }
+        }
+
+        if self.paused && self.declicker.has_settled() {
+            self.damping.reset_to_target();
+            self.room_size.reset_to_target();
+            self.width.reset_to_target();
+
+            return ProcessStatus::ClearAllOutputs;
+        }
+
+        let all_silent = proc_info.in_silence_mask.all_channels_silent(2);
+        if all_silent && self.is_silent {
+            self.declicker.reset_to_target();
+            self.damping.reset_to_target();
+            self.room_size.reset_to_target();
+            self.width.reset_to_target();
+
+            return ProcessStatus::ClearAllOutputs;
+        }
+
+        if !all_silent && self.is_silent {
+            // re-apply the parameters
+            self.apply_parameters();
+            self.is_silent = false;
+        }
+
+        // just take the slow path if any are smoothing
+        if self.damping.is_smoothing() || self.room_size.is_smoothing() || self.width.is_smoothing()
+        {
+            for frame in 0..proc_info.frames {
+                let damping = self.damping.next_smoothed();
+                let room_size = self.room_size.next_smoothed();
+                let width = self.width.next_smoothed();
+
+                // we assume setting these values is more expensive than
+                // calculating their smoothing
+                if frame.is_multiple_of(4) {
+                    self.freeverb.set_dampening(damping as f64);
+                    self.freeverb.set_room_size(room_size as f64);
+                    self.freeverb.set_width(width as f64);
+
+                    self.freeverb.update_combs();
+                }
+
+                let (left, right) = self
+                    .freeverb
+                    .tick((inputs[0][frame] as f64, inputs[1][frame] as f64));
+
+                outputs[0][frame] = left as f32;
+                outputs[1][frame] = right as f32;
+            }
+
+            self.damping.settle();
+            self.room_size.settle();
+            self.width.settle();
+        } else {
+            for frame in 0..proc_info.frames {
+                let (left, right) = self
+                    .freeverb
+                    .tick((inputs[0][frame] as f64, inputs[1][frame] as f64));
+
+                outputs[0][frame] = left as f32;
+                outputs[1][frame] = right as f32;
+            }
+        }
+
+        // We do this before the declicking just to make sure we
+        // finish declicking if we're paused simultaneously with the
+        // input going silent.
+        if all_silent && !self.is_silent {
```

Comment:
would [`check_for_silence_on_outputs`](https://github.com/BillyDM/Firewheel/blob/5eff88dc771a62bac38dbccef7635be8f232d14c/crates/firewheel-core/src/node.rs#L479) work here? (I might be interpreting incorrectly)

Reply:
Yes, I just wrote most of this before updating to the latest version. I should probably update this.